include/ipr/impl

```diff
@@ -1259,6 +1280,7 @@ namespace ipr {
          Template_id* make_template_id(const ipr::Name&,
                                        const ipr::Expr_list&);
          Static_cast* make_static_cast(const ipr::Type&, const ipr::Expr&);
+         Qualification* make_qualification(const ipr::Type&, const ipr::Expr&, Optional<ipr::Type> = {});
```

Comment:
Let's make take two arguments: (1) the source expression; (2) the type of the expression resulting from the cv-qualification.  None of it is optional

Reply:
Type is never optional on an Expr, other make_xxx functions have the constraint as Optional assumably so that it can be filled in later, not emitted altogether.

I was trying to follow this pattern can you provide more context as to why these should differ.

Reply:
The type should be optional only when logically required -- this is not the case here.

The other nodes are for expressions that can occur as dependent expressions, hence the type is unknown therefore optional.  Implicit conversions are always built with known type where they are built.